### PR TITLE
Update logger_factory.py

### PR DESCRIPTION
**Pull request by wmwxwa**
## Why are these changes needed?

Adding Cosmos DB for logging

## Related issue number

N/A

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: autogen/logger/logger_factory.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 from autogen.logger.base_logger import BaseLogger
 from autogen.logger.sqlite_logger import SqliteLogger
+from autogen.logger.cosmos_db_logger import CosmosDBLogger
```
Comment (ekzhu) [inline autogen/logger/logger_factory.py:5]: As mentioned in #2329, we should do this in a try-except block to check for import error as cosmos db is going to be an optional dependency.

**Review by ekzhu — CHANGES_REQUESTED** [review]
Maybe the changes in this PR is better fit into #2329 ?

File: autogen/logger/logger_factory.py [hunk 2]
```diff
@@ -2,9 +2,15 @@
...
```
Comment (cheng-tan) [inline autogen/logger/logger_factory.py:5]: should this line be removed?

**Review by cheng-tan — CHANGES_REQUESTED** [review]

**Comment by wmwxwa** [comment]
> Maybe the changes in this PR is better fit into #2329 ?

Making the change through #2329. Deleting this PR.

**Pull request closed without merging**